### PR TITLE
ADDED: services/micro/c example

### PR DESCRIPTION
**Pull request by levb**
https://nats-by-example-wfyr11i7s-connecteverything.vercel.app/examples/services/intro/c

### CODE REVIEW

File: docker/c/Dockerfile [hunk 1]
```diff
@@ -15,6 +15,7 @@ RUN apt-get install -y libssl-dev
 RUN apt-get install -y wget
 
 COPY install-nats-c.sh .
+COPY CMakeLists.txt .
```
Comment (levb) [inline docker/c/Dockerfile:18]: Will be used to build all C examples

**Review by mtmk — APPROVED** [review]
LGTM

**Review by Jarema — CHANGES_REQUESTED** [review]
Looks ok, and generates the output.

Would be nice to add headers like in Go example:

https://nats-by-example-wfyr11i7s-connecteverything.vercel.app/examples/services/intro/go

**Comment by levb** [comment]
@Jarema I can add headers, but... I don't see headers used in the Go example either. Am I missing something?

**Comment by Jarema** [comment]
@levb sorry, I meant headlines - "Defining a service", "Adding endoints"

**Comment by levb** [comment]
@Jarema Ah! I deliberately removed them since the code vs. the config data are kind of scattered in the C code, no inlines.

**Review by Jarema — APPROVED** [review]
LGTM!

**Pull request merged**